pkg/phases/monitoring/test.go

```diff
@@ -1,12 +1,94 @@
 package monitoring
 
 import (
+	"context"
+	"crypto/tls"
+	"fmt"
 	"github.com/flanksource/commons/console"
 	"github.com/moshloop/platform-cli/pkg/k8s"
 	"github.com/moshloop/platform-cli/pkg/platform"
+	"github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/prometheus/common/model"
+	log "github.com/sirupsen/logrus"
+	"net/http"
+	"time"
+)
+
+const (
+	testMetricName  = "test_metric"
+	testJobName     = "test"
+	waitTimeSeconds = 0
 )
 
 func Test(p *platform.Platform, test *console.TestResults) {
 	client, _ := p.GetClientset()
 	k8s.TestNamespace(client, "monitoring", test)
 }
+
+func TestThanos(p *platform.Platform, test *console.TestResults, args []string) {
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	pushGatewayHost := args[0]
+	thanosHost := args[1]
+	pusher := pushMetric(pushGatewayHost)
+	err := pusher.Add()
+    if err != nil {
+		test.Failf("Thanos: client", "Failed to inject metric to client Prometheus via Pushgateway %v", err)
+		return
+	} else {
+		test.Passf("Thanos client", "Metric successfully injected into the client Prometheus. Waiting to receive it in observability cluster.")
+	}
+	log.Tracef("Waiting for metric")
```

Comment:
Put this into a loop that does the check, and sleeps for 5 seconds up until the wait timeout so that we can exit early

Reply:
I implemented it this way first, but don't we have retry logic already?